smsdk/tool_register.py

```diff
@@ -47,9 +48,13 @@ class SmsdkEntities(metaclass=ABCMeta):
 
     @classmethod
     @abstractmethod
-    def __init__(self, session, base_url) -> None:
-        self.session = session
-        self.base_url = base_url
+    # The subclasses of the SmsdkEntities class will be responsible
+    # for defining their own __init__() methods, which will initialize
+    # the session and base_url attributes.
+    def __init__(self, session: t_.Any, base_url: str) -> None:
```

Comment:
Since I don't see those subclasses changing does this break anthing

Reply:
No, it won't break anything. The rationale behind this change is as follows.

When an abstract method is implemented within the abstract class itself, it ceases to be an abstract method. It transforms into a concrete method, and subclasses are no longer obligated to provide their own implementations for it.

Forcing all subclasses of 'SmsdkEntities' to implement the '__init__' method is the reason I removed its implementation in the abstract class.

Additionally, since this method is decorated as a classmethod, the instance object cannot be accessed as the first parameter within '__init__'. This limitation leads to couple of  static type errors. To resolve these type errors, it's necessary to eliminate the assignment of instance attributes within the '__init__' method.